test/unit/reducer/tests/test-reducer-reset.hpp

```diff
@@ -141,8 +141,8 @@ void testReducerReset()
   reduce_max.reset(resetVal[0]);
   reduce_minloc.reset(resetVal[0]);
   reduce_maxloc.reset(resetVal[0]);
-  reduce_minloctup.reset(resetVal[0]);
   reduce_maxloctup.reset(resetVal[0]);
```

Comment:
@trws Apparently, the `maxloctup.reset()` was getting optimized out in the Sequential version of this test when compiled with NVCC + GCC. Changing the ordering fixes this for some reason, but doing so feels clunky. If there is some better way to get around this, please let me know.

Reply:
Does it get optimized away with GCC only?

Reply:
Right, with GCC only. Not getting this error with other compilers.

Reply:
When you say "optimized out" do you mean the side-effect didn't take effect so it was failing in the assertion below?  That's the kind of thing that really shouldn't happen unless we accidentally have UB somewhere.

Reply:
Right, the check on line 157 fails. It returns the initialized values, or more specifically, the default initialized values.

Reply:
That's a definite cause for concern.  It probably shouldn't matter for sequential anyway, but is it valid to use reset after compute but before get like this? I know we have places where get actually completes the reduction, and can't remember if all of that is wired up right to allow this order.

Reply:
With the cuda and hip backends this should be safe, as reset calls finalizes the reduction by calling get before resetting the value.

Reply:
Other than the author of the test not knowing about calling `get` before `reset`, there is probably no reason for this to happen. 😃

I'll modify the PR to put a `get` call into `reset()`.

Reply:
That's odd, inserting `get` into the `reset` call does not help. Nor does manually calling get in the test itself. I'll still push the changes to `reset`, but keep the test with this modified ordering of calls.